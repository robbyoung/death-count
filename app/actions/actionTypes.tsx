export enum ActionType {
    ADD_PLAYTHROUGH = 'Add Playthrough',
    ADD_GAME = 'Add Game',
    ADD_DEATH = 'Add Death',
    ADD_DEATH_DETAIL = 'Add Death Detail',
    COMPLETE_DEATH = 'Complete Death',
    ADD_OPTION = 'Add Option',
    ADD_OPTION_SET = 'Add Option Set',
    SELECT_GAME = 'Select Game',
    LOAD_STATE = 'Load State',
    COMPLETE_GAME = 'Complete Game',
    SELECT_PLAYTHROUGH = 'Select Playthrough',
    SELECT_OPTION_SET = 'Select Option Set',
    DELETE_PLAYTHROUGH = 'Delete Playthrough',
}
