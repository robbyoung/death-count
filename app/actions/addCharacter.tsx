import { Death, Character } from '../state';
import { Action } from 'redux';

export interface AddCharacterAction extends Action {
    newCharacter: Character;
}

export function addCharacter(state: Character[], action: AddCharacterAction): Character[] {
    return [
        ...state,
        action.newCharacter,
    ]
}