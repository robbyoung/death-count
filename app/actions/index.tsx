export * from './actionTypes';
export * from './addPlaythrough';
export * from './addDeath';
export * from './addGame';
export * from './addDeathDetail';
export * from './completeDeath';