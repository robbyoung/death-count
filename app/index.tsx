import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { Screens } from './screens';
import Home from './containers/home';
import NewDeath from './containers/newDeath';
import GamesList from './containers/gamesList';
import NewGame from './containers/newGame';
import Loading from './containers/loading';
import Startup from './containers/startup';
import PlaythroughsList from './containers/playthroughsList';

const MainNavigator = createStackNavigator(
	{
		Loading: { screen: Loading },
		Home: { screen: Home },
		NewDeath: { screen: NewDeath },
		GamesList: { screen: GamesList },
		NewGame: { screen: NewGame },
		Startup: { screen: Startup },
		PlaythroughsList: { screen: PlaythroughsList }
	},
	{
		initialRouteName: Screens.Loading,
	},
);

export default createAppContainer(MainNavigator);