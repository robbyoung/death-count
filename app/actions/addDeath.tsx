import { Death } from '../state';
import { Action } from 'redux';
import { ActionType } from './actionTypes';

export class AddDeath implements Action {
    type: ActionType = ActionType.ADD_DEATH;

    constructor(public newDeath: Death) {}
}

export function addDeath(state: Death[], action: AddDeath): Death[] {
    return [
        ...state,
        action.newDeath,
    ]
}