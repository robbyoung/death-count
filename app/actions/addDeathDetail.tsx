import { Action } from 'redux';
import { Death } from '../state';
import { ActionType } from './actionTypes';

export interface AddDeathDetailAction extends Action {
    type: ActionType.ADD_DEATH_DETAIL;
    detailKey: string;
    detailValue: string;
}

export function addDeathDetailAction(
    detailKey: string,
    detailValue: string,
): AddDeathDetailAction {
    return {
        type: ActionType.ADD_DEATH_DETAIL,
        detailKey,
        detailValue,
    };
}

export function addDeathDetail(
    state: Death[],
    action: AddDeathDetailAction,
): Death[] {
    const incompleteIndex = state.findIndex(death => !death.complete);
    if (incompleteIndex === -1) {
        return state;
    }

    const newState = [...state];
    const editedDeath: Death = {
        ...state[incompleteIndex],
        details: {
            ...state[incompleteIndex].details,
        },
    };

    editedDeath.details[action.detailKey] = action.detailValue;
    newState[incompleteIndex] = editedDeath;
    return newState;
}
