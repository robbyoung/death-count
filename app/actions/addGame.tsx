import { Game } from '../state';
import { Action } from 'redux';
import { ActionType } from './actionTypes';

export class AddGame implements Action {
    type: ActionType.ADD_GAME;

    constructor(public newGame: Game) {}
}

export function addGame(state: Game[], action: AddGame): Game[] {
    return [
        ...state,
        action.newGame,
    ]
}