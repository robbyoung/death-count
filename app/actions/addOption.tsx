import { OptionSet } from '../state';
import { Action } from 'redux';
import { ActionType } from './actionTypes';

export interface AddOptionAction extends Action {
    type: ActionType.ADD_OPTION;
    id: string,
    value: string,
}

export function addOptionAction(id: string, value: string): AddOptionAction {
    return {
        type: ActionType.ADD_OPTION,
        id,
        value,
    };
}

export function addOption(state: OptionSet[], action: AddOptionAction): OptionSet[] {
    const optionsIndex = state.findIndex(optionSet => optionSet.id === action.id);
    if(state[optionsIndex].options.find(option => option === action.value)) {
        return state;
    }

    const newState = [
        ...state
    ];
    newState[optionsIndex] = {
        ...state[optionsIndex],
        options: [
            ...state[optionsIndex].options,
            action.value,
        ],
    };
    return newState;
}