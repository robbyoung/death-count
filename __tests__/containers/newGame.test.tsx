import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import { addGameAction, addOptionSetAction } from '../../app/actions';
import NewGame from '../../app/containers/newGame';

const fakeNavigation = {
    goBack: () => undefined,
}

describe('New Game Container', () => {
    it('Renders without any option sets', async () => {
        store.dispatch(addGameAction('Test Game'));
        
        const component = renderer.create(<NewGame navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Renders with option sets', async () => {
        const gameAction = addGameAction('Test Game');
        store.dispatch(gameAction);
        store.dispatch(addOptionSetAction('Set 1', gameAction.newGame.id));
        store.dispatch(addOptionSetAction('Set 2', gameAction.newGame.id));
        
        const component = renderer.create(<NewGame navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
