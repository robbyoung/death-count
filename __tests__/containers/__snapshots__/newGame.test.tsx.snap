// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Game Container Renders with option sets 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "fontWeight": "bold",
          "margin": 10,
        }
      }
    >
      Test Game
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderBottomColor": "#592a29",
          "borderBottomWidth": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "margin": 10,
          }
        }
      >
        Set 1
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderBottomColor": "#592a29",
          "borderBottomWidth": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "margin": 10,
          }
        }
      >
        Set 2
      </Text>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Add a new option set"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 10,
            "borderTopLeftRadius": 10,
            "flexGrow": 100,
            "fontSize": 22,
            "height": 50,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": undefined,
              },
              Object {
                "backgroundColor": "#e84641",
                "borderBottomRightRadius": 10,
                "borderTopRightRadius": 10,
                "color": "#ffffff",
                "fontSize": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              },
              Object {
                "fontFamily": "FontAwesome5FreeSolid",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#e84641",
              "borderRadius": 5,
              "color": "#ffffff",
              "fontSize": 30,
              "height": 50,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": "80%",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`New Game Container Renders without any option sets 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "fontWeight": "bold",
          "margin": 10,
        }
      }
    >
      Test Game
    </Text>
    <View
      style={
        Object {
          "alignContent": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Add a new option set"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 10,
            "borderTopLeftRadius": 10,
            "flexGrow": 100,
            "fontSize": 22,
            "height": 50,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": undefined,
              },
              Object {
                "backgroundColor": "#e84641",
                "borderBottomRightRadius": 10,
                "borderTopRightRadius": 10,
                "color": "#ffffff",
                "fontSize": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              },
              Object {
                "fontFamily": "FontAwesome5FreeSolid",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#e84641",
              "borderRadius": 5,
              "color": "#ffffff",
              "fontSize": 30,
              "height": 50,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": "80%",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
